src/notes_manager.py

```diff
@@ -9,10 +6,34 @@ def edit(self, new_content):
         self.content = self.format_note(new_content)
         return f"Note updated to: {self.content}"
 
+    def search_notes_by_tag(self, tag):
```

Comment:
Move all these commands to `Assistant` class and add an attribute: `notes = [None]`